lib/plugins/aws/deploy/lib/createStack.js

```diff
@@ -63,6 +60,13 @@ module.exports = {
           .then(() => BbPromise.resolve('alreadyCreated'))
           .catch(e => {
             if (e.message.indexOf('does not exist') > -1) {
+              if (this.serverless.service.provider.deploymentBucket) {
+                this.createLater = true;
+              }
+
+              if (this.options.noDeploy || this.createLater) {
```

Comment:
because we always need to check whether the stack exists or not even when noDeploy is specified so that we know whether we should create or update, or in the case of noDeploy, whether we should output 1 create template, or both create/update templates